src/storage/btree.c

```diff
@@ -24630,8 +24989,15 @@ btree_range_scan_resume (THREAD_ENTRY * thread_p, BTREE_SCAN * bts)
 
   assert (bts->force_restart_from_root || !VPID_ISNULL (&bts->C_vpid));
   assert (bts->C_page == NULL);
-  assert (!DB_IS_NULL (&bts->cur_key));
   assert (!BTS_IS_INDEX_ILS (bts));
+#if defined(IMPROVE_RANGE_SCAN_IN_BTREE)
+  // TODO: 
+  btree_check_decompress_key (bts);
```

Comment:
What is the purpose of checking for decompression here?

Reply:
When `C_page` cannot be reused, we have to find the location of the previous key again and start scanning.
To find the location of the previous key, only `&bts->cur_key` is passed to the `btree_locate_key` function, so it is necessary to add `common_prefix_key` to `cur_key`.